internal/iapl/policy.go

```diff
@@ -0,0 +1,300 @@
+package iapl
+
+import (
+	"fmt"
+
+	"go.infratographer.com/permissions-api/internal/types"
+)
+
+// PolicyDocument represents a partial authorization policy.
+type PolicyDocument struct {
+	ResourceTypes  []ResourceType
+	TypeAliases    []TypeAlias
+	Actions        []Action
+	ActionBindings []ActionBinding
+}
+
+// ResourceType represents a resource type in the authorization policy.
+type ResourceType struct {
+	Name          string
+	IDPrefix      string
+	Relationships []Relationship
+}
+
+// Relationship represents a named relation between two resources.
+type Relationship struct {
+	Relation       string
+	TargetTypeName string
+}
+
+// TypeAlias represents a named alias to multiple concrete resource types.
+type TypeAlias struct {
+	Name              string
+	ResourceTypeNames []string
+}
+
+// Action represents an action that can be taken in an authorization policy.
+type Action struct {
+	Name string
+}
+
+// ActionBinding represents a binding of an action to a resource type.
+type ActionBinding struct {
+	ActionName       string
+	ResourceTypeName string
+	Conditions       []Condition
+}
+
+// Condition represents a necessary condition for performing an action.
+type Condition struct {
+	RoleBinding        *ConditionRoleBinding
+	RelationshipAction *ConditionRelationshipAction
+}
+
+// ConditionRoleBinding represents a condition where a role binding is necessary to perform an action.
+type ConditionRoleBinding struct {
+}
+
+// ConditionRelationshipAction represents a condition where another action must be allowed on a resource
+// along a relation to perform an action.
+type ConditionRelationshipAction struct {
+	Relation   string
+	ActionName string
+}
+
+// Policy represents an authorization policy as defined by IAPL.
+type Policy interface {
+	Validate() error
+	Schema() []types.ResourceType
+}
+
+var _ Policy = &policy{}
```

Comment:
This doesn't seem like its doing anything?

Reply:
It's making sure `policy` implements `Policy`, but that can be removed.